tests/TestCase/Console/ArgumentsTest.php

```diff
@@ -57,9 +56,43 @@ public function testGetArgumentAt()
     public function testHasArgumentAt()
     {
         $values = ['big', 'brown', 'bear'];
-        $args = new Arguments($values, []);
+        $args = new Arguments($values, [], []);
         $this->assertTrue($args->hasArgumentAt(0));
         $this->assertTrue($args->hasArgumentAt(1));
         $this->assertFalse($args->hasArgumentAt(3));
+        $this->assertFalse($args->hasArgumentAt(-1));
+    }
+
+    /**
+     * check arguments by name
+     *
+     * @return void
+     */
+    public function testHasArgument()
+    {
+        $values = ['big', 'brown', 'bear'];
+        $names = ['size' => 0, 'color' => 1, 'species' => 2, 'odd' => 3];
+        $args = new Arguments($values, [], $names);
+        $this->assertTrue($args->hasArgument('size'));
+        $this->assertTrue($args->hasArgument('color'));
+        $this->assertFalse($args->hasArgument('hair'));
+        $this->assertFalse($args->hasArgument('Hair'), 'casing matters');
+        $this->assertFalse($args->hasArgument('odd'));
+    }
+
+    /**
+     * get arguments by name
+     *
+     * @return void
+     */
+    public function testGetArgument()
+    {
+        $values = ['big', 'brown', 'bear'];
+        $names = ['size' => 0, 'color' => 1, 'species' => 2, 'odd' => 3];
```

Comment:
how happen if  `$name` set like bellow 
```php
$names = ['size', 'color', 'species', 'odd'];
//or 
$names = [0 => 'size', 1 => 'color', 2 => 'species',3 =>  'odd'];
```

Reply:
@saeideng We'd have to flip that data so we can use `isset()`, or switch to using `array_search()`. Using `array_search()` might be simpler for callers now that you mention it.